plugins/modules/aws_api_gateway_domain.py

```diff
@@ -0,0 +1,352 @@
+    # Also this module doesn't support custom TLS cert setup params as they are kind of deprecated already and would increase complexity.
+    existing_domain_settings = {
+        'certificate_arn': domain.get('certificate_arn'),
+        'security_policy': domain.get('security_policy'),
+        'endpoint_type': domain.get('endpoint_configuration').get('types')[0]
+    }
+    specified_domain_settings = {
+        'certificate_arn': module.params.get('certificate_arn'),
+        'security_policy': module.params.get('security_policy'),
+        'endpoint_type': module.params.get('endpoint_type')
+    }
+
+    if specified_domain_settings != existing_domain_settings:
+        try:
+            result['domain'] = update_domain_name(client, domain_name, **snake_dict_to_camel_dict(specified_domain_settings))
+            result['updated'] = True
+        except (ClientError, EndpointConnectionError) as e:
+            module.fail_json_aws(e, msg="updating API GW domain")
+
+    existing_mappings = copy.deepcopy(existing_domain.get('path_mappings', []))
+    # Cleanout `base_path: "(none)"` elements from dicts as those won't match with specified mappings
+    for mapping in existing_mappings:
+        if mapping.get('base_path', 'missing') == '(none)':
+            mapping.pop('base_path')
+
+    specified_mappings = copy.deepcopy(module.params.get('domain_mappings', []))
+    # Cleanout `base_path: ""` elements from dicts as those won't match with existing mappings
+    for mapping in specified_mappings:
+        if mapping.get('base_path', 'missing') == '':
+            mapping.pop('base_path')
+
+    if specified_mappings != existing_mappings:
+        try:
+            # When lists missmatch delete all existing mappings before adding new ones as specified
+            for mapping in existing_domain.get('path_mappings', []):
+                delete_domain_mapping(client, domain_name, mapping['base_path'])
+            for mapping in module.params.get('domain_mappings', []):
+                result['path_mappings'] = add_domain_mapping(
+                    client, domain_name, mapping.get('base_path', ''), mapping.get('rest_api_id'), mapping.get('stage')
+                )
+                result['updated'] = True
+        except (ClientError, EndpointConnectionError) as e:
+            module.fail_json_aws(e, msg="updating API GW domain mapping")
+
+    return camel_dict_to_snake_dict(result)
+
+
+def delete_domain(module, client):
+    domain_name = module.params.get('domain_name')
+    try:
+        result = delete_domain_name(client, domain_name)
+    except (ClientError, EndpointConnectionError) as e:
+        module.fail_json_aws(e, msg="deleting API GW domain")
+    return camel_dict_to_snake_dict(result)
+
+
+retry_params = {"tries": 10, "delay": 5, "backoff": 1.2}
+
+
+@AWSRetry.backoff(**retry_params)
+def get_domain_name(client, domain_name):
+    return client.get_domain_name(domainName=domain_name)
```

Comment:
I guess we need to enclose all client.something() calls in a try catch block.

Reply:
Shouldn't there be a global meachnism of outputting errors if this is really required everywhere. 

Also I don't really have to energy to go through yet another round of refactoring of this module without any hope that this will ever get merged (despite 3 code reviews in the past that I all adressed already/gave greenlight). So I would suggest to merge the PR now (with all the other changes I already addressed) or close it for good.